osquery/tables/networking/darwin/wifi.mm

```diff
@@ -140,17 +140,15 @@ QueryData genKnownWifiNetworks(QueryContext& context) {
     }
   } else if (CFGetTypeID(networks) == CFDictionaryGetTypeID()) {
     auto count = CFDictionaryGetCount((CFDictionaryRef)networks);
-    auto keys = static_cast<const void **>(malloc(sizeof(void *) * count));
-    auto values = static_cast<const void **>(malloc(sizeof(void *) * count));
-
-    if (keys != nullptr && values != nullptr) {
-      CFDictionaryGetKeysAndValues((CFDictionaryRef)networks, keys, values);
-      for (CFIndex i = 0; i < count; i++) {
-        parseNetworks((CFDictionaryRef)values[i], results);
-      }
+    std::vector<const void *> keys;
```

Comment:
Check out the `3`rd overloaded constructor: http://en.cppreference.com/w/cpp/container/vector/vector